src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs

```diff
@@ -77,21 +78,31 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
+            IDisposable linePragmaScope = null;
             if (node.Source != null)
             {
-                using (Context.Writer.BuildLinePragma(node.Source.Value))
-                {
-                    var padding = BuildOffsetPadding(0, node.Source.Value, Context);
-                    Context.Writer.Write(padding);
+                linePragmaScope = Context.Writer.BuildLinePragma(node.Source.Value);
+                var padding = BuildOffsetPadding(0, node.Source.Value, Context);
+                Context.Writer.Write(padding);
+            }
 
-                    Context.AddLineMappingFor(node);
-                    Context.Writer.Write(node.Content);
+            for (var i = 0; i < node.Children.Count; i++)
+            {
+                if (!(node.Children[i] is RazorIRToken token) || !token.IsCSharp)
+                {
+                    continue;
                 }
+
+                Context.AddLineMappingFor(token);
+                Context.Writer.Write(token.Content);
```

Comment:
Take a look at https://github.com/aspnet/Razor/pull/1115/files#diff-8df250e114e6022d67d79cdd0e5c4b56L39

This handles extension nodes as well.